Update to new user API.

mdwelsh opened the pull request:
This PR updates the `fixie` package to use the new `/api/v1/user/me` API for getting and updating details on the current user. This is needed so we can support this in the web UI.

mdwelsh commented:
@petersalas This is now ready for a review. 

I am making some important structural changes here: The build goes in 'dist', and getting rid of 'web.ts' which is no longer needed as far as I can tell. Can you please look?

petersalas commented on packages/fixie/.gitignore:
```diff
@@ -1,2 +0,0 @@
-*.js
-*.d.ts
```
why did you remove this?

petersalas commented on packages/fixie/tsconfig.json:
```diff
@@ -9,10 +9,11 @@
     "lib": ["dom", "dom.iterable", "ES2022"]
   },
-  "include": ["src/*", "jest.config.ts", ".eslintrc.cjs", "index.ts", "web.ts"],
+  "include": ["src/**/*", "jest.config.ts", ".eslintrc.cjs", "index.ts"],
```
I'm pretty sure you're breaking into jail w.r.t. ESM/CJS/TS because Next.js apps default to CJS -- at the very least you should do this as part of a different change

petersalas commented on packages/fixie/src/index.ts:
```diff
@@ -0,0 +1,3 @@
+export * from './client.js';
+export * from './fixie-embed.js';
```
I'm not sure about combining `fixie` and `fixie/web` because it will require React everywhere

mdwelsh commented:
Okay! Thanks for the comments. Reverted everything but the specific change to the user stuff, but also renamed sidekick-types.ts to types.ts since I put the User type in there too.

petersalas commented on packages/fixie/src/agent.ts:
```diff
@@ -500,7 +500,7 @@ export class FixieAgent {
   ): Promise<AgentRevision> {
     const config = await FixieAgent.LoadConfig(agentPath);
-    const agentId = `${(await client.userInfo()).username}/${config.handle}`;
+    const agentId = config.handle;
```
This is a misleading variable name -- can you remove the variable and just use `config.handle` everywhere?

petersalas commented on packages/fixie/src/agent.ts:
```diff
@@ -549,7 +549,7 @@ export class FixieAgent {
   }) {
     const config = await FixieAgent.LoadConfig(agentPath);
-    const agentId = `${(await client.userInfo()).username}/${config.handle}`;
+    const agentId = config.handle;
```
Ditto

petersalas commented on packages/fixie/src/main.ts:
```diff
@@ -161,6 +163,19 @@ program
+  .command('update')
+  .description('Update information on the current user')
+  .option('--email <string>', 'The new email address for this user')
+  .option('--fullName <string>', 'The new description for this corpus')
```
bad copy/paste in the arg help

mdwelsh commented:
### Merge activity

* **Nov 9, 9:47 PM**: @mdwelsh merged this pull request with [Graphite](https://app.graphite.dev/github/pr/fixie-ai/ai-jsx/487).
